### ISSUE TITLE
Elements triggering Presentational Roles Conflict Resolution

### ISSUE THREAD

**Issue by giacomo-petri** [auto-wcag/auto-wcag#2278]
## Accessibility support

>Implementation of [Presentational Roles Conflict Resolution](https://www.w3.org/TR/wai-aria-1.1/#conflict_resolution_presentation_none) varies from one browser or assistive technology to another. Depending on this, some [semantic](https://www.w3.org/WAI/standards-guidelines/act/rules/23a2a8/#semantic-role) img elements can fail this rule with some technology but users of other technologies would not experience any accessibility issue.

and

> Implementation of the [Presentational Roles Conflict Resolution](https://www.w3.org/TR/wai-aria-1.1/#conflict_resolution_presentation_none) differs slightly from one user agent to the other. Hence, some elements might be exposed by one user agent and not by another, and consequently might create accessibility issues only for some users. Nevertheless, triggering the conflict is a bad practice.

## ACT Rules

- [Image has non-empty accessible name - 23a2a8](https://www.w3.org/WAI/standards-guidelines/act/rules/23a2a8/proposed/#accessibility-support)
- [Element marked as decorative is not exposed - 46ca7f](https://www.w3.org/WAI/standards-guidelines/act/rules/46ca7f/proposed/#accessibility-support)

### Test case 1

Focusable (non-nevative tabindex computed value) elements with presentational role
```
<button role="presentation">Test</button>
<a href="#" role="presentation">Test</a>
<h1 role="presentation" tabindex="0">Test</h1>
<button role="presentation" tabindex="-1">Test</button>
<a href="#" role="presentation" tabindex="-1">Test</a>
<h1 role="presentation" tabindex="-1">Test</h1>
<img src="some_src" role="presentation" tabindex="0">
```

[Test cases](https://codepen.io/giacomo-usablenet/full/bNbPJWo)

#### Test instructions

- Just run the test and check the element role

### External links

- WPT tests: https://wpt.fyi/results/wai-aria/role/role_none_conflict_resolution.html?label=experimental&label=master&aligned
- WAI-ARIA 1.1: https://www.w3.org/TR/wai-aria-1.1/#conflict_resolution_presentation_none
- WAI-ARIA 1.2: https://www.w3.org/TR/wai-aria-1.2/#conflict_resolution_presentation_none

**Comment by giacomo-petri** [comment]
## Test results

ARIA spec covers presentational roles conflict and browsers are consistently supporting that spec. 

```md
## Test case 1

- Date: 01 Jan 2024
- Operating system: not relevant
- Browser: Safari, Chrome, Firefox
- Assistive technology: not relevant
- Test result: Passed
```

**Comment by shunguoy** [comment]
The purpose of the rules is to report the root cause of an issue properly so that the author can fix it, and browsers or ATs can handle the content consistently to minimize confusion.   
   1. The "none" or "presentation" role can only be used in certain elements based on the [Aria in HTML spec](https://www.w3.org/TR/html-aria/).  For example, it is invalid to use the "none" or "presentation" role in <button> or <a> with a href.
   For the <img> example, <img src="some_src" role="presentation" tabindex="0">, the root cause of the issue is that the "presentation" role is not valid in this case. The only valid role for the <img> with neither an alt attribute nor an accessible name is the implicit "img" role. 
  
Those issues should be caught by this [ACT rule](https://www.w3.org/WAI/standards-guidelines/act/rules/ff89c9/proposed/). 

2. In the case <img src="some_src" alt="" tabindex="0">, the <img> has an implicit role "presentation", but it's focusable, causing an inconsistent issue with an AT and keyboard.  No ACT rule covers this case. An [ACT rule](https://www.w3.org/WAI/standards-guidelines/act/rules/6cfa84/) covers the case with aria-hidden.

**Comment by giacomo-petri** [comment]
Hi @shunguoy,

During the F2F meeting we discussed and put in place a procedure to better track the accessibility support notes so we can easily maintain them.

This issue is not meant to say that this is good. A presentational conflict role is by itself an author error and violation. The fact that browsers are solving this consistently, doesn't mean that it's fine.

I've opened this GitHub issue as an "accessibility support issue" because before there were some ambiguity in how browsers were handling those scenarios and we want to track which is their behaviour.
